File: app/src/main/java/com/example/federico/mlibrefedericopuy/network/RetrofitClient.java
package com.example.federico.mlibrefedericopuy.network;

import android.content.Context;

import okhttp3.OkHttpClient;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class RetrofitClient {
    private static Retrofit retrofit = null;

    public static ApiInterface getClient() {

        if (retrofit==null){
            OkHttpClient client = new OkHttpClient.Builder()
                    .build();

            retrofit = new Retrofit.Builder()
                    .baseUrl(NetworkUtils.COMPLETEURL)
                    .client(client)
                    .addConverterFactory(GsonConverterFactory.create())
                    .build();
        }
        return retrofit.create(ApiInterface.class);
    }
}

File: app/src/main/java/com/example/federico/mlibrefedericopuy/network/RetrofitClient.kt
package com.example.federico.mlibrefedericopuy.network

import okhttp3.OkHttpClient
import retrofit2.Retrofit
import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
import retrofit2.converter.gson.GsonConverterFactory

object RetrofitClient {
    private var retrofit: Retrofit? = null

    val client: ApiInterface
        get() {
            if (retrofit == null) {
                val client = OkHttpClient.Builder()
                        .build()

                retrofit = Retrofit.Builder()
                        .baseUrl(NetworkUtils.COMPLETEURL)
                        .client(client)
                        .addConverterFactory(GsonConverterFactory.create())
                        .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
                        .build()
            }
            return retrofit!!.create(ApiInterface::class.java)
        }
}
